Source/OxyPlot/Rendering/TextArranger.cs

```diff
@@ -0,0 +1,259 @@
+            out string[] lines,
+            out ScreenPoint[] linePositions)
+        {
+            if (text == null)
+            {
+                throw new ArgumentNullException(nameof(text));
+            }
+
+            // measure the size of the whole text
+            var wholeSize = this.MeasureText(text, fontFamily, fontSize, fontWeight);
+
+            // infer font metrics from wholeSize
+            var metrics = this.TextMeasurer.GetFontMetrics(fontFamily, fontSize, fontWeight);
+            var lineHeight = metrics.CellHeight;
+            var leading = metrics.Leading;
+            var descender = metrics.Descender;
+
+            // get the rendering bounds from maxSize if necessary
+            var bounds = wholeSize;
+            if (maxSize != null)
+            {
+                bounds = new OxySize(Math.Ceiling(Math.Min(bounds.Width, maxSize.Value.Width)), Math.Ceiling(Math.Min(bounds.Height, maxSize.Value.Height)));
+            }
+
+            // split into lines
+            lines = StringHelper.SplitLines(text);
+
+            // if the text is too tall, we need to remove some lines
+            if (bounds.Height < wholeSize.Height && lines.Length > 1)
+            {
+                var clippedLineCount = 1 + (int)((bounds.Height - lineHeight) / (lineHeight + leading));
+                lines = lines.Take(clippedLineCount).ToArray();
+
+                if (this.SquashTrimmedTextBounds)
+                {
+                    bounds = new OxySize(bounds.Width, ((lineHeight + leading) * lines.Length) - leading);
+                }
+            }
+
+            // if the text is too wide, we need to trim the lines down a bit
+            if (bounds.Width < wholeSize.Width)
+            {
+                for (int i = 0; i < lines.Length; i++)
+                {
+                    lines[i] = this.TextTrimmer.Trim(this.TextMeasurer, lines[i], bounds.Width, fontFamily, fontSize, fontWeight);
+                }
+
+                if (this.SquashTrimmedTextBounds)
+                {
+                    bounds = new OxySize(lines.Max(l => this.TextMeasurer.MeasureTextWidth(l, fontFamily, fontSize, fontWeight)), bounds.Height);
+                }
+            }
+
+            // do these once
+            var sin = Math.Sin(rotation / 180.0 * Math.PI);
+            var cos = Math.Cos(rotation / 180.0 * Math.PI);
+
+            // turn metrics into vectors
+            var offsetBoundsWidth = new ScreenVector(cos * bounds.Width, sin * bounds.Width);
+            var offsetBoundsHeight = new ScreenVector(-sin * bounds.Height, cos * bounds.Height);
+
+            var offsetLineHeight = new ScreenVector(-sin * lineHeight, cos * lineHeight);
+            var offsetLeading = new ScreenVector(-sin * leading, cos * leading);
+            var offsetDescender = new ScreenVector(-sin * descender, cos * descender);
+
+            // align to bounds
+            var offsetBoundsX = offsetBoundsWidth * 0.0; // keep centerline
```

Comment:
`offsetBoundsX` is always a (0,0)-vector?! Also this is the only usage of `offsetBoundsWidth`, so its value has no meaning? Am I missing something?

Reply:
I don't think you're missing anything... I'd better take a proper look at it.